tests/e2e-kubernetes/testsuites/upgrade.go

```diff
@@ -0,0 +1,376 @@
+		deferCleanup(func(ctx context.Context) error { return e2epod.DeletePodWithWait(ctx, f.ClientSet, pod) })
+
+		return pod
+	}
+
+	checkBasicFileOperations := func(pod *v1.Pod) {
+		seed := time.Now().UTC().UnixNano()
+		filename := fmt.Sprintf("test-%d.txt", seed)
+		path := filepath.Join(e2epod.VolumeMountPath1, filename)
+		testWriteSize := 1024 // 1KB
+
+		checkWriteToPath(f, pod, path, testWriteSize, seed)
+		checkReadFromPath(f, pod, path, testWriteSize, seed)
+		checkListingPathWithEntries(f, pod, e2epod.VolumeMountPath1, []string{filename, "test.txt"})
+		checkDeletingPath(f, pod, path)
+		checkListingPathWithEntries(f, pod, e2epod.VolumeMountPath1, []string{"test.txt"})
+	}
+
+	updateCSIDriversServiceAccountRole := func(ctx context.Context, oidcProvider, policyName string) {
+		By("Updating CSI Driver's Service Account Role for IRSA")
+		sa := csiDriverServiceAccount(ctx, f)
+
+		role, removeRole := createRole(ctx, f, assumeRoleWithWebIdentityPolicyDocument(ctx, oidcProvider, sa), policyName)
+		deferCleanup(removeRole)
+
+		sa, restoreServiceAccountRole := overrideServiceAccountRole(ctx, f, sa, *role.Arn)
+		deferCleanup(restoreServiceAccountRole)
+
+		waitUntilRoleIsAssumableWithWebIdentity(ctx, f, sa)
+
+		// Trigger recreation of our pods to use the new IAM role
+		killCSIDriverPods(ctx, f)
+	}
+
+	assignPolicyToServiceAccount := func(ctx context.Context, sa *v1.ServiceAccount, policyName string) *v1.ServiceAccount {
+		role, removeRole := createRole(ctx, f, assumeRoleWithWebIdentityPolicyDocument(ctx, oidcProvider, sa), policyName)
+		deferCleanup(removeRole)
+
+		sa, _ = overrideServiceAccountRole(ctx, f, sa, *role.Arn)
+		waitUntilRoleIsAssumableWithWebIdentity(ctx, f, sa)
+		return sa
+	}
+
+	createServiceAccountWithPolicy := func(ctx context.Context, policyName string) *v1.ServiceAccount {
+		sa, removeSA := createServiceAccount(ctx, f)
+		deferCleanup(removeSA)
+
+		return assignPolicyToServiceAccount(ctx, sa, policyName)
+	}
+
+	enablePLI := func(ctx context.Context) context.Context {
+		return contextWithVolumeAttributes(ctx, map[string]string{"authenticationSource": "pod"})
+	}
+
+	BeforeEach(func(ctx context.Context) {
+		l = local{}
+		l.config = driver.PrepareTest(ctx, f)
+		DeferCleanup(cleanup)
+	})
+
+	It("Upgrade to a new version from a previous version without interrupting the workloads", func(ctx context.Context) {
+		if helmChartPreviousVersion == "" || helmChartNewVersion == "" {
+			Fail("Please set the previous and new versions to test using `MOUNTPOINT_CSI_DRIVER_PREVIOUS_VERSION` and `MOUNTPOINT_CSI_DRIVER_NEW_VERSION` environment variables")
+		}
+
+		oidcProvider = oidcProviderForCluster(ctx, f)
+		if oidcProvider == "" {
+			Fail("Please configure OIDC provider for the testing cluster")
+		}
+
+		framework.Logf("Testing upgrade from %q to %q...", helmChartPreviousVersion, helmChartNewVersion)
+
+		settings, cfg := initHelmClient()
+
+		// Make sure to start from a clean state
+		uninstallCSIDriver(cfg)
```

Comment:
Just to make sure: would it make sense to also uninstall the EKS ad-don here to make sure it's in a clean state? I'm not sure how unclean state could happen so don't know if this makes sense.